app/src/main/java/com/stevesoltys/seedvault/ui/storage/SafStorageOptions.kt

```diff
@@ -50,6 +54,49 @@ internal class SafStorageOptions(
         roots.add(root)
     }
 
+    /**
+     * Add a storage root for each child directory at the RoundSync root, if it exists.
+     */
+    private fun checkOrAddRoundSyncRoots(roots: ArrayList<SafOption>) {
+
+        val roundSyncRoot = roots.firstOrNull {
+            it.authority == AUTHORITY_ROUND_SYNC
+        } ?: return
+
+        roots.remove(roundSyncRoot)
+
+        val childrenUri = DocumentsContract.buildChildDocumentsUriUsingTree(
+            roundSyncRoot.uri, roundSyncRoot.documentId
+        )
+        val projection = arrayOf(COLUMN_DISPLAY_NAME, COLUMN_DOCUMENT_ID)
+        val cursor = context.contentResolver.query(childrenUri, projection, null, null, null)
+
+        cursor?.use {
+            val nameIndex = cursor.getColumnIndex(COLUMN_DISPLAY_NAME)
+            val documentIdIndex = cursor.getColumnIndex(COLUMN_DOCUMENT_ID)
+
+            while (cursor.moveToNext()) {
+                val name = cursor.getString(nameIndex)
+                val documentId = cursor.getString(documentIdIndex)
+
+                val childRoot = SafOption(
+                    authority = AUTHORITY_ROUND_SYNC,
+                    rootId = name,
+                    documentId = documentId,
+                    icon = getIcon(context, AUTHORITY_ROUND_SYNC, name, 0),
+                    title = name,
+                    summary = context.getString(R.string.storage_round_sync_summary_prefix) + name,
+                    availableBytes = null,
```

Comment:
The Roundsync provider doesn't expose this? Would be nice to know when dealing with out of space situations.

Reply:
Not that I'm aware.